sample/src/main/java/com/polidea/cockpit/sample/apipresentation/ApiPresentationBaseFragment.kt

```diff
@@ -0,0 +1,48 @@
+package com.polidea.cockpit.sample.apipresentation
+
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.util.TypedValue
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.polidea.cockpit.sample.R
+import kotlinx.android.synthetic.main.fragment_api_presentation.*
+
+abstract class ApiPresentationBaseFragment<T : ApiPresentationBaseContract.Presenter> : Fragment(), ApiPresentationBaseContract.View<T> {
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
+            inflater.inflate(R.layout.fragment_api_presentation, container, false)
+
+
+    override fun showFooter(isVisible: Boolean) {
+        footer_container.visibility = if (isVisible) View.VISIBLE else View.INVISIBLE
+    }
+
+    override fun setTextColor(color: Int) {
+        cockpit_textview.setTextColor(color)
+        cockpit_color_textview.setTextColor(color)
+    }
+
+    override fun setFontSize(textSize: Float) {
+        cockpit_textview.setTextSize(TypedValue.COMPLEX_UNIT_SP, textSize)
+    }
+
+    override fun setColorDescription(description: String) {
+        cockpit_color_textview.text = description
+    }
+
+    override fun setFooterText(footerText: String) {
+        footer_text_view.text = footerText
+    }
+
+    override fun onResume() {
+        super.onResume()
+        presenter.start()
```

Comment:
I suppose you would like to init views and add callback listeners in `presenter.start()` method. If so, the method should be called in `onViewCreated()` method (it similar to Activity's `onCreate`). Then presenter's method should be named `onViewCreated()` also.

Reply:
It shouldn't not be called onViewCreated(). Presenter should not be dependent on a view lifecycle.